### ISSUE TITLE
Broken code

### ISSUE THREAD

**Issue by Knguyen21** [ga-wdi-boston/full-stack-project#164]
Hi, 

I tried adding some code on my data and now I can't display any meals on my web page. I tried going back to the original code, but it's still broken. I would like to have someone look over my code .

**Comment by GA-MEB** [comment]
What things were finished and working just before things broke?
What were you working on when you made the breaking change?
What code did you add, and where?

**Comment by Knguyen21** [comment]
I fixed it, but I'm having another issue. I'm having trouble getting the attribute id of a handlebar to jquery.

**Comment by GA-MEB** [comment]
Could you rephrase that? I'm not sure what you mean.

**Comment by jrhorn424** [comment]
@Knguyen21 If you're having trouble explaining your problem, it might be a sign that you are going to fast, or that you are changing too many things at once. Please provide more context and code examples so @MattB-GA-Boston can think of a debugging strategy.

**Comment by Knguyen21** [comment]
I'm trying to display the reviews for certain meals by using the meal id . However, I'm having trouble doing it.  Each meals have a review button. I'm trying to set an id to the review button that will match with the meal id. In addition I need to send an ajax request with the meal id in the url

**Comment by GA-MEB** [comment]
@Knguyen21, you're saying that you have a list of meals, each with a review button? And when you click that button, you want to see a list of reviews? Is that right?

**Comment by jrhorn424** [comment]
I believe she wants to show a form when you click "Review".

**Comment by Knguyen21** [comment]
yes. all the reviews for the specific meal

**Comment by GA-MEB** [comment]
OK. So if each meal has its own 'Review'/'Reviews' button, one thing you could do is store the id of each meal as a data attribute on the meal, and reference that meal's data attribute from the button.

Could you paste in the HTML/Handlebars code for a meal?

**Comment by Knguyen21** [comment]
```
<script>console.log('meal-listing.hb start');</script>
{{#each meals}}
{{id}}
<br>{{name}}
<p>{{desc}}</p>
{{price}}
<br>
<button type="button" class="btn btn-primary btn-sm" data-toggle="modal" data-target="#reviews" data-meal-id="{{id}}">
  Reviews</button>

{{/each}}
<script>console.log('meal-listing.hb end');</script>
```

**Comment by GA-MEB** [comment]
Cool! So, what if you were to put each of these meals inside a `div`, and set `meal-id` as a data attribute? If you did that, you could refer back to the `div` from the button (since it would be the parent element) and use that to grab the value of `meal-id`.

**Comment by GA-MEB** [comment]
e.g.

``` handlebars
{{#each meals}}
<div data-meal-id={{id}}>
  {{id}} 
  <br>
  {{name}}
  <p>{{desc}}</p>
  {{price}}
  <br>
  <button type="button" class="btn btn-primary btn-sm" data-toggle="modal" data-target="#reviews" data-meal-id="{{id}}">
  Reviews</button>
</div>
{{/each}}
```

**Comment by gaand** [comment]
handlebars has a built in logger, see http://handlebarsjs.com/builtin_helpers.html

**Comment by Knguyen21** [comment]
I'm having issues with the ajax request url. 

``` js
// edited by @jrhorn424 - A newline after backticks is required. A file extension is helpful, too.
let getReviews= function(e){
  let meal_id = require('./meal-listing.handlebars');
  console.log('meal id' + meal_id);
  $.ajax({


    url: myApp.baseURL + "/meals/" + $(e.target).attr('data-meal-id')+ "/reviews",
    method: 'GET',
    dataType: 'json'
  }).done(function(response){
    console.log('Get request');
    displayReviews(response,'#userReviews');
  });
};

let displayReviews= function(response, review){
  let reviews = response.reviews;
     console.log("yes reviews");
    let reviewListingTemplate = require('./review-listing.handlebars');
        $(review).html(reviewListingTemplate({
            reviews
          }));

  console.log(response.meals);
};

$("#data-meal-id").on('click',function(){
  getReviews();
});
```

**Comment by GA-MEB** [comment]
Are you sure that `e.target` is referring to the `div`? If this handler is responding to a button click, the target would be the button, not the button's parent element. Try `console.log`ing `e.target` - make sure you're grabbing the right element.

Also, for data attributes, jQuery has a special method that you should use.

**Comment by jrhorn424** [comment]
@MattB-GA-Boston @Knguyen21 When this is resolved, please add any notes from your discussion and close. Thanks so much!

**Comment by Knguyen21** [comment]
Thank you! The function that handles the click handler was not working because the elements that I tried to attach the handler to did not exist yet because they were part of a handlebars template that had not been rendered yet.
